crates/miden-testing/tests/scripts/faucet.rs

```diff
@@ -662,3 +668,205 @@ async fn network_faucet_burn() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+// TESTS FOR MINT NOTE WITH PRIVATE AND PUBLIC OUTPUT MODES
+// ================================================================================================
+
+/// Tests creating a MINT note with private output mode (8 inputs)
+#[tokio::test]
+async fn test_mint_note_private_output_mode() -> anyhow::Result<()> {
+    let mut builder = MockChain::builder();
+
+    let faucet_owner_account_id = AccountId::dummy(
+        [1; 15],
+        AccountIdVersion::Version0,
+        AccountType::RegularAccountImmutableCode,
+        AccountStorageMode::Private,
+    );
+
+    let faucet =
+        builder.add_existing_network_faucet("NET", 1000, faucet_owner_account_id, Some(50))?;
+    let target_account = builder.add_existing_wallet(Auth::IncrNonce)?;
+
+    let amount = Felt::new(75);
+    let mint_asset: Asset = FungibleAsset::new(faucet.id(), amount.into()).unwrap().into();
+    let aux = Felt::new(27);
+    let serial_num = Word::default();
```

Comment:
Nit: I think it would be worth deduplicating the test setup between the private and public test, since this is around 50 lines of code each.

Reply:
In the interest of unblocking, I think it's fine to use `add_note_script` instead and this should get solved mostly automatically once #2036 is addressed.